### PR TITLE
doc: add pull request docs

### PR DESCRIPTION
**Pull request by matthiasdiener**
here is how it looks like rendered: https://mirgecom.readthedocs.io/en/doc-pr/development.html

### CODE REVIEW

File: doc/development.rst [hunk 1]
```diff
@@ -37,6 +37,164 @@ The source repository (and current branch) of each of these packages
 in use is determined by the file
 `requirements.txt in mirgecom <https://github.com/illinois-ceesd/mirgecom/blob/master/requirements.txt>`__.
 
+
+Working with Pull Requests
+--------------------------
+
+We are using GitHub's pull requests (PRs) feature to integrate changes into
+mirgecom. Pull requests are based on git branches that are merged into another
```
Comment (inducer) [inline doc/development.rst:45]:
```suggestion
mirgecom and its supporting packages. Pull requests are based on git branches that are merged into another
```

File: doc/development.rst [hunk 2]
```diff
@@ -37,6 +37,164 @@ The source repository (and current branch) of each of these packages
...
+^^^^^
+
+Apart from the `git` tool, there are other tools that help to simplify various
+aspects of working with GitHub:
+
+Command line:
+~~~~~~~~~~~~~
+
+-  https://hub.github.com/
+-  https://github.com/cli/cli
```
Comment (inducer) [inline doc/development.rst:188]: Make these proper links.

File: doc/development.rst [hunk 3]
```diff
@@ -37,6 +37,164 @@ The source repository (and current branch) of each of these packages
...
+
+GUI
+~~~
+
+-  Fork
+-  GitHub Desktop
+-  Sublime Merge
```
Comment (inducer) [inline doc/development.rst:195]: Make these proper links

Comment (inducer) [inline doc/development.rst:195]: Add Magit. (I'm saying that as a Vim person!)

File: doc/development.rst [hunk 4]
```diff
@@ -37,6 +37,164 @@ The source repository (and current branch) of each of these packages
...
+
+6. After the pull request has been merged, you can delete the branch
+   (locally and remotely):
+
+   .. code:: bash
+
+      $ git branch -d featureX    # delete branch locally
+      $ git push –delete featureX # delete it remotely; or delete in web interface
+
+Updating a pull request
```
Comment (inducer) [inline doc/development.rst:121]: Add a blurb about dependent pull requests. Here's a start:

> It is expected that you will need to propose changes to supporting packages (e.g. grudge, meshmode, loopy, pyopencl), depending on ...

File: doc/development.rst [hunk 5]
```diff
@@ -37,6 +37,164 @@ The source repository (and current branch) of each of these packages
...
 
+
+Working with Pull Requests
+--------------------------
+
+We are using GitHub's pull requests (PRs) feature to integrate changes into
+mirgecom. Pull requests are based on git branches that are merged into another
+branch (usually the master branch). Note that pull requests are a GitHub
+feature and live outside the main git functionality; the `git` program itself
+has no knowledge of them.
```
Comment (inducer) [inline doc/development.rst:48]: Link to https://try.github.io/

Comment (inducer) [inline doc/development.rst:48]: Mention that `conda install git` can be used to install git. (or that emirge automatically installs it?)

File: doc/development.rst [hunk 6]
```diff
@@ -37,6 +37,164 @@ The source repository (and current branch) of each of these packages
...
+force-push it back to the repository (i.e, ``git push --force``). Please
+use this sparingly.
+
+Reviewing/CI
+^^^^^^^^^^^^
+
+Each pull requests for mirgecom needs one manual approval by a reviewer and
+needs to pass the Continuous Integration (CI) tests before merging. We use
+GitHub actions as the CI provider to test each pull request. The CI tests are
+triggered automatically when a pull request is created or updated.
```
Comment (inducer) [inline doc/development.rst:142]: Describe how to navigate the CI output (what is Flake8?), what is supposed to happen if a test fails (they fix it), how to mark PRs as draft, and how to request reviews. (Example of a PR that could have used this: https://github.com/inducer/pyrometheus/pull/2)

**Review by inducer — COMMENTED** [review]
Thanks! This is super important, and likely very helpful in the long run. Some thoughts below.

File: doc/development.rst (hunk 4 above)
Comment (matthiasdiener) [reply doc/development.rst:131]: Done.

File: doc/development.rst (hunk 6 above)
Comment (matthiasdiener) [reply doc/development.rst:157]: Done.

File: doc/development.rst [hunk 7]
```diff
@@ -37,6 +37,212 @@ The source repository (and current branch) of each of these packages
...
+      # Install these with 'pip install wheel flake8 pep8-naming flake8-quotes pytest pylint'
+      $ flake8
+      $ pydocstyle
+      $ pytest
+
+3. Commit your changes to the new branch:
+
+   .. code:: bash
+
+      $ git commit
```
Comment (inducer) [inline doc/development.rst:93]:
```suggestion
      $ git add file1.py file2.py
      $ git commit
```

File: doc/development.rst [hunk 8]
```diff
@@ -37,6 +37,212 @@ The source repository (and current branch) of each of these packages
...
+Creating a new pull request
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Create a new branch for your pull request:
+
+   .. code:: bash
+
+      $ cd mirgecom           # or loopy, meshmode, ...
+      $ git branch featureX   # Create new branch for your feature
+      $ git checkout featureX # Activate the branch
```
Comment (inducer) [inline doc/development.rst:77]:
```suggestion
      $ git switch -c featureX   # Create new branch for your feature and switch to it
```

File: doc/development.rst [hunk 9]
```diff
@@ -37,6 +37,212 @@ The source repository (and current branch) of each of these packages
...
+      # Edit files
+
+2. Check that your changes pass tests:
+
+   .. code:: bash
+
+      # Install these with 'pip install wheel flake8 pep8-naming flake8-quotes pytest pylint'
```
Comment (inducer) [inline doc/development.rst:84]:
```suggestion
      # Install these with 'pip install wheel flake8 pep8-naming flake8-quotes pytest pylint' if needed
      # (emirge will have installed them for you automatically)
```

File: doc/development.rst [hunk 10]
```diff
@@ -37,6 +37,212 @@ The source repository (and current branch) of each of these packages
...
+      $ flake8
+      $ pydocstyle
+      $ pytest
```
Comment (inducer) [inline doc/development.rst:87]:
```suggestion
      $ flake8 mirgecom test
      $ pydocstyle mirgecom test
      $ (cd test && python -m pytest)
```

File: doc/development.rst [hunk 11]
```diff
@@ -37,6 +37,212 @@ The source repository (and current branch) of each of these packages
...
+   On most systems, the ``git`` program will already be installed. You can
+   install it yourself by running ``conda install git``.
+
+Forking the repository
+^^^^^^^^^^^^^^^^^^^^^^
+
+In most cases, it will be necessary to fork the repository (that is, create a
+copy of the repository in your own GitHub account), since creating a pull
+request requires creating a branch in the repo, which requires write access.
+You can fork a repository directly on the GitHub website.
```
Comment (inducer) [inline doc/development.rst:66]: This should cover adding a remote for the forked repo. (Used under the name `my_fork` below.)

File: doc/development.rst [hunk 12]
```diff
@@ -37,6 +37,212 @@ The source repository (and current branch) of each of these packages
...
+4. Push your changes:
+
+   .. code:: bash
+
+      $ git push -u
+      # alternatively:
+      $ git push -u origin featureX
+
+   If you do not have write privileges to this repository, you can push
+   the change to your fork instead:
```
Comment (inducer) [inline doc/development.rst:104]:
```suggestion
```
(The folks who are allowed to push directly already know how to do it.)

File: doc/development.rst [hunk 13]
```diff
@@ -37,6 +37,212 @@ The source repository (and current branch) of each of these packages
...
+
+   .. code:: bash
+
+      $ git push -u <forked_repo> featureX
```
Comment (inducer) [inline doc/development.rst:108]:
```suggestion
      $ git push -u my_fork featureX
```

File: doc/development.rst [hunk 14]
```diff
@@ -37,6 +37,212 @@ The source repository (and current branch) of each of these packages
...
+   https://github.com/illinois-ceesd/mirgecom/pulls
+
+   The `base` branch should be the `master` branch of the repo you want to
+   merge into in most cases. The `compare` branch is the branch with your
+   changes.
+
+   When creating the pull request, select at least one reviewer (someone that
+   has knowledge about the code you are modifying), add yourself as the
+   assignee, and choose appropriate labels (if any). Note that this can
+   usually not be done for a PR from a forked repository.
```
Comment (inducer) [inline doc/development.rst:121]:
```suggestion
```

File: doc/development.rst [hunk 15]
```diff
@@ -37,6 +37,212 @@ The source repository (and current branch) of each of these packages
...
+to express this dependency to make it easier to review and test both PRs jointly.
+
+You can express this dependency by modifying the branch of a dependent package
+inside mirgecom's ``requirements.txt`` file in the main mirgecom folder. In
+the following example, assume that we want to create a feature in mirgecom
+that depends on the ``my_branch`` branch in meshmode::
+
+   git+https://github.com/inducer/meshmode.git#egg=meshmode
+   # change to:
+   git+https://github.com/inducer/meshmode.git@my_branch#egg=meshmode
```
Comment (inducer) [inline doc/development.rst:162]:
```suggestion
   git+https://github.com/MYUSERNAME/meshmode.git@featureX#egg=meshmode
```

**Review by inducer — COMMENTED** [review]
Nice! A few more suggestions/thoughts below.

File: doc/development.rst (hunk 11 above)
Comment (matthiasdiener) [reply doc/development.rst]: Good idea, done.

File: doc/development.rst [hunk 16]
```diff
@@ -37,6 +37,205 @@ The source repository (and current branch) of each of these packages
...
+
+In most cases, it will be necessary to fork the repository (that is, create a
+copy of the repository in your own GitHub account), since creating a pull
+request requires creating a branch in the repo, which requires write access.
+You can fork a repository directly on the GitHub website. You can add the URL of
+the fork to your local cloned copy of the repository with the following command:
+
+.. code:: bash
+
+   $ git remote add my_fork git@github.com:MYUSERNAME/<repo>
```
Comment (inducer) [inline doc/development.rst:71]:
```suggestion
   $ git remote add my_fork git@github.com:MYUSERNAME/REPONAME
```
(Only use one style for "meta" variables)

**Review by inducer — APPROVED** [review]
:+1:

**Review by inducer — APPROVED** [review]
Thx!

**Pull request merged**